tartufo/scanner.py

```diff
@@ -32,19 +34,108 @@
 PatternDict = Dict[str, Union[str, Pattern]]
 
 
-class Bcolors:
-    HEADER = "\033[95m"
-    OKBLUE = "\033[94m"
-    OKGREEN = "\033[92m"
-    WARNING = "\033[93m"
-    FAIL = "\033[91m"
-    ENDC = "\033[0m"
-    BOLD = "\033[1m"
-    UNDERLINE = "\033[4m"
-
-    def __init__(self, name):
-        # type: (str) -> None
-        self.name = name
+class IssueType(enum.Enum):
+    Entropy = "High Entropy"
+    RegEx = "Regular Expression Match"
+
+
+class Issue:
+    """Represents an issue found while scanning the code."""
+
+    OUTPUT_SEPARATOR = "~~~~~~~~~~~~~~~~~~~~~"  # type: str
+    DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S"  # type: str
+
+    issue_type = None  # type: Optional[IssueType]
+    issue_detail = None  # type: Optional[str]
+    diff = None  # type: Optional[git.Diff]
+    strings_found = None  # type: Optional[List[str]]
+    commit = None  # type: Optional[git.Commit]
+    branch_name = None  # type: Optional[str]
+
+    def __init__(self, issue_type, strings_found):
+        # type: (IssueType, List[str]) -> None
+        self.issue_type = issue_type
+        self.strings_found = strings_found
+
+    def as_dict(self):
+        # type: () -> Dict[str, Optional[str]]
+        """Return a dictionary representation of an issue.
+
+        This is primarily meant to aid in JSON serialization.
+        """
+        output = {
+            "issue_type": self.issue_type.value,  # type: ignore
+            "issue_detail": self.issue_detail,
+            "diff": self.printable_diff,
+            "strings_found": self.strings_found,
+            "commit_time": self.commit_time,
+            "commit_message": self.commit_message,
+            "commit_hash": self.commit_hash,
+            "file_path": self.file_path,
+            "branch": self.branch_name,
+        }
+        return output
+
+    @property
+    def printable_diff(self):
+        # type: () -> str
+        if not self.diff:
+            return "No diff available."
+        return self.diff.diff.decode("utf-8", errors="replace")
+
+    @property
+    def commit_time(self):
+        # type: () -> Optional[str]
+        if not self.commit:
+            return None
+        commit_time = datetime.datetime.fromtimestamp(self.commit.committed_date)
+        return commit_time.strftime(self.DATETIME_FORMAT)
+
+    @property
+    def commit_message(self):
+        # type: () -> Optional[str]
+        if not self.commit:
+            return None
+        return self.commit.message
+
+    @property
+    def commit_hash(self):
+        # type: () -> Optional[str]
+        if not self.commit:
+            return None
+        return self.commit.hexsha
```

Comment:
Is it possible to do ternary here?
```return None if not self.commit else self.commit.hexsha```?

Reply:
It is possible to do that, yes. I've just always found that syntax to be rather less readable. While it is certainly more compact, I find it less easy to mentally parse.